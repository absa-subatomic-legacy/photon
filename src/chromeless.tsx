import { Chromeless } from 'chromeless';
// tslint:disable-next-line
const { toMatchImageSnapshot } = require('jest-image-snapshot');
import * as fs from 'fs';

export const setup = () => {
  expect.extend({ toMatchImageSnapshot });
  // tslint:disable-next-line: no-any
  setTimeout((global as any).config.timeout);
  // tslint:disable-next-line: no-any
  return new Chromeless((global as any).config.chromeless);
};

// tslint:disable-next-line: no-any
export const teardown = async (chromeless: Chromeless<any>) => {
  try {
    await chromeless.end();
  } catch (err) {
    // tslint:disable-next-line
    console.log(err);
  }
};

export const getFile = (path: string) => Promise.resolve(fs.readFileSync(path));
