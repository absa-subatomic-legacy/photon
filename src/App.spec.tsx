import { setup, teardown } from './chromeless';

describe('When visiting the homepage', function() {
  // tslint:disable-next-line: no-any
  let chromeless: any;

  beforeAll(() => {
    chromeless = setup();
  });

  afterAll(async function() {
    teardown(chromeless);
  });

  // test('it welcomes the user', async function () {
  //     const html = await chromeless
  //         .goto('https://www.google.com')
  //         .type('chromeless', 'input[name="q"]')
  //         .press(13)
  //         .wait('#resultStats')
  //         .evaluate(() => document!.querySelector('#resualtStats')!.innerHTML);

  //     expect(html).toMatchSnapshot();
  // });

  // test('home renders correctly (visual)', async () => {
  //     const screenshotPath: string = await chromeless
  //         // tslint:disable-next-line: no-any
  //         .goto((global as any).config.baseUrl)
  //         .wait('.App-intro')
  //         .screenshot();

  //     const screenshot = await getFile(screenshotPath);
  //     // tslint:disable-next-line
  //     (expect(screenshot) as any).toMatchImageSnapshot();
  // });

  test('home renders correctly', async function() {
    const html = await chromeless
      // tslint:disable-next-line: no-any
      .goto((global as any).config.baseUrl)
      .wait('#root')
      .evaluate(() => document!.querySelector('#root')!.innerHTML);

    expect(html).toMatchSnapshot();
  });

  test('can navigates to the Profile page via Link', async function() {
    const html = await chromeless
      // tslint:disable-next-line: no-any
      .goto((global as any).config.baseUrl)
      .wait('#root')
      .click('button[data-test="goToProfile"]')
      .wait('.hello')
      .evaluate(() => document!.querySelector('.hello')!.innerHTML);

    expect(html).toMatchSnapshot();
  });

  test('can navigates to the Profile page via Redux', async function() {
    const html = await chromeless
      // tslint:disable-next-line: no-any
      .goto((global as any).config.baseUrl)
      .wait('#root')
      .click('button[data-test="goToProfile"]')
      .wait('.hello')
      .evaluate(() => document!.querySelector('.hello')!.innerHTML);

    expect(html).toMatchSnapshot();
  });

  // test('can fork the project and display it on home page', async function () {
  //     const html = await chromeless
  //         // tslint:disable-next-line: no-any
  //         .goto(`${(global as any).config.baseUrl}/profile/react`)
  //         .wait('.hello')
  //         // Click on Fork the Project button
  //         .exists('[data-test="forkTheProject"]')
  //         .click('[data-test="forkTheProject"]')
  //         .wait(1000)
  //         // Click on Home link to navigates to the Home page
  //         .click('[data-test="linkHome"]')
  //         .evaluate(() => document!.querySelector('.hello')!.innerHTML);

  //     expect(html).toMatchSnapshot();
  // });

  // test('can fork the project and display it on home page', async function () {
  //     const html = await chromeless
  //         // tslint:disable-next-line: no-any
  //         .goto((global as any).config.baseUrl)
  //         .wait('#root')
  //         // Click on Got To Profile button
  //         .click('button[data-test="goToProfileViaRedux"]')
  //         // Wait redirect to Profile page and find the hello class
  //         .wait(1000)
  //         .exists('.hello')
  //         // Click on Fork the Project button
  //         .exists('button[data-test="forkTheProject"]')
  //         // .click('button[data-test="forkTheProject"]')
  //         // .wait(1000)
  //         .evaluate(() => document.querySelector('.hello')!.innerHTML);
  //         // // Click on Home link to navigates to the Home page
  //         // .click('button[data-test="linkHome"]')
  //         // // Find the ProfileView about forked project on Home page
  //         // .wait(1000)
  //         // .wait('.project')
  //         // .evaluate(() => document!.querySelector('.project')!.innerHTML);

  //     expect(html).toMatchSnapshot();
  // });
});
