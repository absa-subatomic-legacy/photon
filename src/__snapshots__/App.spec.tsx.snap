// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When visiting the homepage can navigates to the Profile page via Link 1`] = `"<p class=\\"greeting\\">Hello! My name is <strong>react</strong>. My name was passed in through a route param!</p><button data-test=\\"forkTheProject\\">Fork the Project</button>"`;

exports[`When visiting the homepage can navigates to the Profile page via Redux 1`] = `"<p class=\\"greeting\\">Hello! My name is <strong>react</strong>. My name was passed in through a route param!</p><button data-test=\\"forkTheProject\\">Fork the Project</button>"`;

exports[`When visiting the homepage home renders correctly 1`] = `"<div class=\\"App\\"><header class=\\"App-header\\"><img src=\\"/static/media/logo.5d5d9eef.svg\\" class=\\"App-logo\\" alt=\\"logo\\"><h1 class=\\"App-title\\">Welcome to React</h1></header><p class=\\"App-intro\\"><a data-test=\\"linkHome\\" href=\\"/\\">Home</a>&nbsp;&nbsp;<a data-test=\\"linkProfile\\" href=\\"/profile/react\\">Profile</a></p><section><p>Welcome to the React Photon Application.<br>The entire apps build using React!<br>Check out our docs on <a href=\\"https://reactjs.org\\">reactjs.org</a> to get started.</p><a href=\\"/profile/react\\"><button data-test=\\"goToProfile\\">Go to profile page via Link</button></a><button data-test=\\"goToProfileViaRedux\\">Go to profile page via redux</button></section></div>"`;
